builtin/providers/aws/resource_aws_cloudformation_stack.go

```diff
@@ -69,10 +73,14 @@ func resourceAwsCloudFormationStack() *schema.Resource {
 				Computed: true,
 			},
 			"policy_body": &schema.Schema{
-				Type:      schema.TypeString,
-				Optional:  true,
-				Computed:  true,
-				StateFunc: normalizeJson,
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateJsonString,
+				StateFunc: func(v interface{}) string {
+					json, _ := normalizeJsonString(v)
+					return json
```

Comment:
Good point. I thought about factoring this out, but then I have a wish to replace this with an appropriate `DiffSuppressFunc` function. Therefore, I would consider these not-there-to-stay long term. Also, I would hope that the compiler is going to inline these... :)